src/app/ubs-admin/components/ubs-admin-table/ubs-admin-table.component.html

```diff
@@ -0,0 +1,97 @@
+    <li *ngFor="let column of responsiblePerson; let i = index">
+      <input type="checkbox" value="id1" checked (change)="changeColumns(column.field, i + 24)" />{{ column.field }}
+    </li>
+    <button class="btn btn-info" (click)="showAllColumns()">Clear</button>
+  </ul>
+  <button type="button" class="btn btn-default" class="export-excel-btn">Експортувати в Excel</button>
+</div>
+<div class="scrolling">
+  <table *ngIf="!isLoading; else spinner" mat-table [dataSource]="dataSource" matSort class="mat-elevation-z0" cdkDropListGroup>
+    <ng-container *ngFor="let column of columns; let i = index" [matColumnDef]="column.field" [sticky]="column.sticky" class="column.field">
+      <mat-header-cell
+        *matHeaderCellDef
+        cdkDropList
+        cdkDropListLockAxis="x"
+        cdkDropListOrientation="horizontal"
+        (cdkDropListDropped)="dropListDropped($event, i)"
+        cdkDrag
+        (cdkDragStarted)="dragStarted($event, i)"
+        [cdkDragData]="{ name: column.field, columIndex: i }"
+      >
+        <div *ngIf="column.field !== 'select'; else checkbox">
+          {{ column.field }}
+          <mat-icon *ngIf="arrowDirection !== column.field" (click)="getSortingDate(column.field, 'asc')">arrow_downward</mat-icon>
+          <mat-icon *ngIf="arrowDirection === column.field" (click)="getSortingDate(column.field, 'desc')">arrow_upward</mat-icon>
+        </div>
+        <ng-template #checkbox>
+          <mat-checkbox
+            (change)="$event ? masterToggle() : null"
+            [ngStyle]="{ backgroundColor: '#fff', height: '16px' }"
+            [checked]="selection.hasValue() && isAllSelected()"
+            [indeterminate]="selection.hasValue() && !isAllSelected()"
+            [aria-label]="checkboxLabel()"
+          >
+          </mat-checkbox>
+        </ng-template>
+      </mat-header-cell>
+      <mat-cell *matCellDef="let row" [ngStyle]="column.style" class="column.field">
+        <div *ngIf="column.field !== 'select'; else checkbox">
+          <span *ngIf="column.field === 'orderDate'">{{ row[column.field] | date: 'dd/MM/yyyy' }}</span>
+          <span *ngIf="column.field !== 'orderDate'">{{ row[column.field] }}</span>
+        </div>
+        <ng-template #checkbox class="checkbox">
+          <mat-checkbox
+            (click)="$event.stopPropagation()"
+            (change)="$event ? selection.toggle(row) : null"
+            [checked]="selection.isSelected(row)"
+            [aria-label]="checkboxLabel(row)"
+          >
+          </mat-checkbox>
+        </ng-template>
+      </mat-cell>
+    </ng-container>
+    <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
+    <mat-row *matRowDef="let row; columns: displayedColumns"></mat-row>
+  </table>
+</div>
+<div>
+  <mat-paginator [pageSizeOptions]="[2, 5, 10]" showFirstLastButtons></mat-paginator>
+</div>
+<ng-template #spinner>
+  <!-- TODO: use our Greenciry spinner -->
+  <!-- <mat-spinner></mat-spinner> -->
```

Comment:
it should be another spinner
the one we have in a shared folder
write me, so i can guide you about this

Reply:
Created shared module and added spinner component for all modules